Utils/ActorUtils.cs

```diff
@@ -0,0 +1,55 @@
+﻿using MMRando.Models.Rom;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace MMRando.Utils
+{
+    public class ActorUtils
+    {
+        public static List<Actor> GetSceneActorsByNumber(int SceneNumber, int ActorNumber)
+        {
+            List<Actor> SearchActor = new List<Actor>();
+            List<Actor> SceneActors;
+            // search the scene list starting from the requested scene looking backwards
+            // SceneList follows the internal entrance list but skips some
+            // so the desired scene should be found at or before the requested index
+            for (int i = Math.Min(RomData.SceneList.Count, SceneNumber); i >= 0; i--)
+            {
+                if (RomData.SceneList[i].Number == SceneNumber)
```

Comment:
Just use a lambda expression here
`var scene = RomData.SceneList.Single(x => x.Number == sceneNumber)`

Actually, a lot of this function could be simplified with lambdas

Reply:
Haven't encountered Single before. Is it smart enough to search from the back? Thinking on it, that's more of a rhetorical question. The compiler is always smarter than the developer. I'll definitely swap over to Single. lambdas are still relatively new, and bafflingly powerful.